components/support/error/src/lib.rs

```diff
@@ -1,79 +1,77 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
 /// Define a wrapper around the the provided ErrorKind type.
 /// See also `define_error` which is more likely to be what you want.
+
 #[macro_export]
 macro_rules! define_error_wrapper {
     ($Kind:ty) => {
-        /// Re-exported, so that using crate::error::* gives you the .context()
-        /// method, which we don't use much but should *really* use more.
-        pub use failure::ResultExt;
-
         pub type Result<T, E = Error> = std::result::Result<T, E>;
+        use std::fmt;
+        use std::fmt::{Debug, Display};
+        struct Context<T: Display + Sync + 'static> {
```

Comment:
It's probably better to remove the generic parameter, rename this to ErrorData or something, and replace `context: T` with `kind: ErrorKind`.